src/payload/payloadMsg.ts

```diff
@@ -0,0 +1,127 @@
+import { SignerSupplier } from '../core/builders';
+import { BytesEncodingStatus, PayloadType } from '../core/enums';
+import { BaseMessage, Message, Msg } from '../core/message';
+import { AllPayloads, UnencodedActionPayload } from '../core/payload';
+import { AnySignatureType, Signature, SignatureType } from '../core/signature';
+import { bytesToBase64 } from '../utils/base64';
+import { objects } from '../utils/objects';
+import { kwilEncode } from '../utils/rlp';
+import { bytesToHex } from '../utils/serial';
+
+export interface PayloadMsgOptions {
+  challenge: string;
+  signatureType: AnySignatureType;
+  identifier: Uint8Array;
+  signer: SignerSupplier;
+  signature: Signature<BytesEncodingStatus.BASE64_ENCODED>;
+}
+
+/**
+ * `PayloadMsg` class creates a call message payload that can be sent over GRPC.
+ */
+export class PayloadMsg {
+  public payload: AllPayloads;
```

Comment:
Does this actually need to be `AllPayloads`? I don't think so...

You're only going to use this for actions.